src/sagemaker/session.py

```diff
@@ -4196,6 +4202,21 @@ def _deployment_entity_exists(describe_fn):
         return False
 
 
+def _endpoint_config_name_exists(describe_fn):
```

Comment:
This function is 90% duplicated from _deployment_entity_exists. We should eliminate the duplication.